### PR TITLE
[json-rpc] Add `preburn_queues` field to DesignatedDealer view

### PR DESCRIPTION
**Pull request by tzakian**
This adds a new field to the `DesignatedDealer` variant of the `AccountRoleView` struct. We make sure to preserve backwards compatibility by setting the old `preburn_balances` field to be the sum of all preburn balances in the preburn queue for each currency.

Additionally, to further decouple the off-chain changes from the on-chain state changes, even if a DD hasn't been upgraded yet to the new preburn queue on chain, we fill in the `preburn_queues` field with the current preburns for the DD if it hasn't switched yet.

### CODE REVIEW

File: json-rpc/tests/integration_test.rs [hunk 1]
```diff
@@ -205,7 +205,18 @@ fn create_test_cases() -> Vec<Test> {
...
+                                    "amount": 0_u64,
+                                    "currency": "XUS"
+                                }
+                            ],
+                            "preburn_queues": [
+                                {
+                                    "preburns": [],
+                                    "currency": "XUS",
+                                }
+                            ],
```
Comment (xli) [inline json-rpc/tests/integration_test.rs:219]: Could we add a test that produces non-zero preburn_balances info? it's fine if we could not produce preburn_queues yet, but from my understanding, we at least should get same result in preburn_queues with preburn_balances, before we enable the feature.
Not sure whether we can have test with feature enabled, it would be great to have it too.

The diff looks solid, but I trust more on test result than my eyes.

Comment (tzakian) [reply json-rpc/tests/integration_test.rs:219]: Sounds good. I'll work on adding that test right now. We should indeed get the same result in both of those fields. 

Currently we can only create preburn queues (so with the feature enabled) since it is running on the head of the framework.

Comment (tzakian) [reply json-rpc/tests/integration_test.rs:219]: Just added the test. Added it as another integration test, and made sure that the `preburns_balances` value is equal to the sum of the preburn requests in the preburn queue.

**Review by xli — APPROVED** [review]

File: types/src/account_state.rs [hunk 2]
```diff
@@ -114,9 +131,23 @@ impl AccountState {
...
+                    Ok(preburn_queues),
+                    Ok(Some(designated_dealer)),
+                ) => {
+                    let preburn_balances =
+                        if preburn_balances.is_empty() && !preburn_queues.is_empty() {
+                            DesignatedDealerPreburns::PreburnQueue(preburn_queues)
+                        } else if !preburn_balances.is_empty() && preburn_queues.is_empty() {
+                            DesignatedDealerPreburns::Preburn(preburn_balances)
+                        } else {
+                            return Ok(None);
```
Comment (moeziniaf) [inline types/src/account_state.rs:149]: nice

**Review by moeziniaf — APPROVED** [review]
thank you!

File: json-rpc/types/src/views.rs [hunk 3]
```diff
@@ -68,9 +69,63 @@ pub enum AccountRoleView {
         received_mint_events_key: BytesView,
         compliance_key_rotation_events_key: BytesView,
         base_url_rotation_events_key: BytesView,
+        preburn_queues: Option<Vec<PreburnQueueView>>,
```
Comment (tzakian) [inline json-rpc/types/src/views.rs:72]: @xli I wound up having to change this field to be nullable in order to keep backwards compatibility (see the cluster test failure -- new client, but talking to an older json-rpc server). Let me know if the change looks good to you.

Comment (bmwill) [reply json-rpc/types/src/views.rs:72]: Yeah, that's correct. Any fields added to an existing struct or variant would need to be nullable. for compatibility reasons.
